### PR TITLE
eventhandler: Refactor to eliminate global state saver

### PR DESCRIPTION
**Pull request by vsiddharth**
Signed-off-by: Vinothkumar Siddharth <sidvin@amazon.com>

### Summary

Removes global state saver from eventhandler

### Testing

- [x] Builds on Linux (`make release`)
- [x] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [x] Unit tests on Linux (`make test`) pass
- [x] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [x] Integration tests on Linux (`make run-integ-tests`) pass
- [x] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [x] Functional tests on Linux (`make run-functional-tests`) pass
- [x] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes:  NO

### Licensing

This contribution is under the terms of the Apache 2.0 License:  YES

### CODE REVIEW

File: agent/eventhandler/task_handler.go [hunk 1]
```diff
@@ -53,6 +54,10 @@ type TaskHandler struct {
 	// * taskToEvents
 	// * tasksToContainerStates
 	taskHandlerLock sync.RWMutex
+
+	// statesaver is a statemanager which may be used to save any
+	// changes to a task or container's SentStatus
+	statesaver statemanager.Saver
```
Comment (aaithal) [inline agent/eventhandler/task_handler.go]: lint: `stateSaver`

File: agent/eventhandler/task_handler.go [hunk 2]
```diff
@@ -61,6 +66,7 @@ func NewTaskHandler() *TaskHandler {
 		tasksToEvents:          make(map[string]*eventList),
 		submitSemaphore:        utils.NewSemaphore(concurrentEventCalls),
 		tasksToContainerStates: make(map[string][]api.ContainerStateChange),
+		statesaver:             statemanager.NewNoopStateManager(),
```
Comment (aaithal) [inline agent/eventhandler/task_handler.go]: Am I missing something here or do you need an additional change to set `statesaver` to the actual state saver object created in `app/agent.go` ?

Comment (vsiddharth) [reply agent/eventhandler/task_handler.go]: I'd made an incorrect assumption here which was clarified post discussion.

File: agent/eventhandler/task_handler.go (hunk 1 above)
Comment (vsiddharth) [reply agent/eventhandler/task_handler.go]: Done

**Comment by vsiddharth** [comment]
@aaithal I've pushed another revision. Could you please take another look when you have some time ?

**Pull request merged**